soh/soh/Enhancements/debugger/hookDebugger.cpp

```diff
@@ -0,0 +1,111 @@
+#include "hookDebugger.h"
+#include "../game-interactor/GameInteractor.h"
+#include "../../UIWidgets.hpp"
+#include <string>
+#include <version>
+
+static std::unordered_map<const char*, std::unordered_map<HOOK_ID, HookInfo>> hookData;
+
+const ImVec4 grey = ImVec4(0.75, 0.75, 0.75, 1);
+const ImVec4 yellow = ImVec4(1, 1, 0, 1);
+const ImVec4 red = ImVec4(1, 0, 0, 1);
+
+void DrawHookRegisteringInfos(const char* hookName) {
+    if (hookData[hookName].size() == 0) {
+        ImGui::TextColored(grey, "No hooks found");
+        return;
+    }
+
+    if (ImGui::BeginTable(
+        ("Table##" + std::string(hookName)).c_str(),
+        4,
+        ImGuiTableFlags_Resizable | ImGuiTableFlags_Reorderable | ImGuiTableFlags_Hideable | ImGuiTableFlags_Borders | ImGuiTableFlags_SizingFixedFit
```

Comment:
Are you satisfied with the buffer spaces at the end of tables that don't have entries long enough to fit the window width? SizingStretchProp or Same would probably size the tables to fill the window, with the side effect of the Registration Info maybe not showing everything, but the window could always be resized if necessary.